### PR TITLE
Implement verification on transparency log

### PR DESCRIPTION
**Pull request by tiainen**
## Description

Fixes pyrsia/pyrsia#687

This PR adds verification on the transparency log of the downloaded docker manifest and docker blobs. Verification happens by comparing the computed hash on the downloaded bytes against the hash that is available in the transparency log.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and everything passes.
- [x] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

File: src/docker/v2/handlers/blobs.rs [hunk 1]
```diff
@@ -143,6 +149,23 @@ pub async fn handle_put_blob(
         .unwrap())
 }
 
+fn verify_blob_content(
```
Comment (fishseabowl) [inline src/docker/v2/handlers/blobs.rs:152]: Add some unit tests if possible. Thanks

**Review by fishseabowl — COMMENTED** [review]
I like the Arc<Mutex<...>> type. I hope it improves performance.

**Comment by prince-chrismc** [comment]
Its really worrying that our test code is not being ran 🤔 there might be some other issue impacting whats happening in the CI... maybe a reporting bug?

image

I think we need a separate issue to investigate this

File: src/docker/v2/handlers/blobs.rs [hunk 2]
```diff
@@ -20,13 +20,16 @@ use crate::docker::docker_hub_util::get_docker_hub_auth_token;
 use crate::docker::error_util::{RegistryError, RegistryErrorCode};
 use crate::docker::v2::storage::*;
 use crate::network::client::{ArtifactType, Client};
+use crate::transparency_log::log::TransparencyLogError;
 use bytes::Bytes;
+use easy_hasher::easy_hasher::raw_sha256;
```
Comment (fishseabowl) [inline src/docker/v2/handlers/blobs.rs:25]: JFYI: [multihash](https://github.com/multiformats/multihash) might be better.

**Review by efrisch — DISMISSED** [review]
lgtm.

**Review by betarelease — DISMISSED** [review]

**Review by erwin1 — APPROVED** [review]

**Review by jperedadnr — APPROVED** [review]
Looks good

**Pull request merged**